### PR TITLE
[11.0] [ADD] queue_nsca_client

### PR DESCRIPTION
**Pull request by etobella**
This addon allows to send information to your nsca server.

Configure
=========

* Access with developer mode
* Access *Settings / NSCA Client / Checks*
* Create all the checks you need for your queue system

The check must have the model *queue.job*, the method *cron_queue_status* and
the following arguments:
- domain: Domain to check (required)
- created_seconds: will only count the jobs older thant this seconds
- critical: Number of records that will mark the critical
- warning: Number of records that will mark the warning

For example, the arguments *[('state', '=', 'failed')], created_seconds=5, critical=2, warning=1*
will return ok if no jobs older than 5 seconds are failed, warning if only one
and otherwise it will return a critical.

The performance data will be the number of jobs found by the domain.

It depends on OCA/server-tools#1280.

### CODE REVIEW

**Comment by guewen** [comment]
Interesting. I don't use nsca addons but seems legit. Just a thought: wouldn't make sense for the OCA to have a repository for all the NSCA addons (nsca_client and children)?

**Comment by etobella** [comment]
It could be a good idea if more people are interested on nsca addons.
@sebalix

File: oca_dependencies.txt [hunk 1]
```diff
@@ -1,15 +1,2 @@
...
-# sale-workflow
-#
-# To explicitely give the URL of a fork, and still use the version specified in
-# .travis.yml, use:
-# sale-workflow https://github.com/OCA/sale-workflow
-#
-# To provide both the URL and a branch, use:
-# sale-workflow https://github.com/OCA/sale-workflow branchname
+server-tools
+server-tools-nsca https://github.com/etobella/server-tools 11.0-mig-nsca_client
```
Comment (pedrobaeza) [inline oca_dependencies.txt]: Is this still needed?

File: queue_nsca_client/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "website": "http://github.com/OCA/server-tools",
+    "author": "Creu Blanca, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": [
+        'queue_job',
+        'nsca_client',
+    ],
+    "data": [
```
Comment (pedrobaeza) [inline queue_nsca_client/__manifest__.py]: Nitpicking: not needed

File: queue_nsca_client/models/queue_job.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+class QueueJob(models.Model):
+    """ Job status and result """
+    _inherit = 'queue.job'
+
+    @api.model
+    def cron_queue_status(self, domain, created_seconds=False,
+                          warning=False, critical=False):
+        if created_seconds:
+            date = datetime.now() + timedelta(seconds=-created_seconds)
```
Comment (pedrobaeza) [inline queue_nsca_client/models/queue_job.py]: Use `fields.Datetime.from_string(fields.Datetime.now)` for tz consistency.

File: queue_nsca_client/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -0,0 +1,16 @@
+* Access with developer mode
+* Access *Settings / NSCA Client / Checks*
+* Create all the checks you need for your queue system
+
+The check must have the model *queue.job*, the method *cron_queue_status* and
+the following arguments:
+- domain: Domain to check (required)
```
Comment (pedrobaeza) [inline queue_nsca_client/readme/CONFIGURE.rst:8]: Extra line for proper RST formatting

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Little things

**Comment by jarroyomorales** [comment]
@etobella Any progress with this?

**Comment by etobella** [comment]
@pedrobaeza @jarroyomorales Comments attended

**Comment by guewen** [comment]
> wouldn't make sense for the OCA to have a repository for all the NSCA addons (nsca_client and children)?

Any opinion on this @etobella @pedrobaeza @jarroyomorales ?

PSC-wise, it would make sense to have NSCA-addons grouped so the actors using NSCA can take care of their addons in one repository.

File: queue_nsca_client/models/queue_job.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+class QueueJob(models.Model):
+    """ Job status and result """
+    _inherit = 'queue.job'
+
+    @api.model
+    def cron_queue_status(self, domain, created_seconds=False,
+                          warning=False, critical=False):
+        if created_seconds:
+            date = fields.Datetime.from_string(
+                fields.Datetime.now) + timedelta(seconds=-created_seconds)
```
Comment (jarroyomorales) [inline queue_nsca_client/models/queue_job.py]: missing parenthesis, `fields.Datetime.now()`

**Comment by jarroyomorales** [comment]
> > wouldn't make sense for the OCA to have a repository for all the NSCA addons (nsca_client and children)?
> 
> Any opinion on this @etobella @pedrobaeza @jarroyomorales ?
> 
> PSC-wise, it would make sense to have NSCA-addons grouped so the actors using NSCA can take care of their addons in one repository.

I think it makes sense, +1

**Comment by jarroyomorales** [comment]
@guewen @etobella @pedrobaeza If a nsca repo is not going to be created, can we try to get this module merged here? Thank you!

**Pull request closed without merging**